native-authentication-sample/src/main/java/com/okta/sample/auth/MainActivity.kt

```diff
@@ -84,31 +87,39 @@ fun FormUi(form: Form) {
             .fillMaxHeight()
     ) {
         for (element in form.elements) {
-            when (element) {
-                is Element.Action -> {
-                    Button(onClick = element.onClick) {
-                        Text(text = element.text)
-                    }
-                }
-                is Element.Label -> {
-                    val fontSize = when (element.type) {
-                        Element.Label.Type.DESCRIPTION -> 16
-                        Element.Label.Type.HEADER -> 20
-                    }
-                    Text(text = element.text, fontSize = fontSize.sp)
-                }
-                is Element.TextInput -> {
-                    TextInputUi(element)
-                }
+            ElementUi(element)
+        }
+    }
+}
+
+@Composable
+fun ElementUi(element: Element) {
+    when (element) {
+        is Element.Action -> {
+            Button(onClick = element.onClick) {
+                Text(text = element.text)
             }
         }
+        is Element.Label -> {
+            val fontSize = when (element.type) {
+                Element.Label.Type.DESCRIPTION -> 16
+                Element.Label.Type.HEADER -> 20
+            }
+            Text(text = element.text, fontSize = fontSize.sp)
+        }
+        is Element.TextInput -> {
+            TextInputUi(element)
+        }
+        is Element.Options -> {
+            OptionsUi(element)
+        }
     }
 }
 
 @Composable
 fun TextInputUi(element: Element.TextInput) {
-    var text by rememberSaveable { mutableStateOf(element.value) }
-    var passwordVisible by rememberSaveable { mutableStateOf(false) }
+    var text by remember { mutableStateOf(element.value) }
+    var passwordVisible by remember { mutableStateOf(false) }
```

Comment:
This is probably just me being a noob with compose... I don't really want to remember anything! I just want to use the mutable state of stuff.

Let me mess with it to see if that's what I can come up with.

We probably need the remember for the password visible case. But the rest of them, the state is held in the element. So the state is mostly just to provide a mechanism to trigger compose to update.

I'll either post another comment here, or push a change.

Reply:
Ok, I looked into how this works a little more. I think we want remember for most things, but since this state is outside of the element, we want `rememberSaveable` which I updated the commit to use. This seems right now from what I can tell!

Reply:
Thanks for looking into this!